qemu/tests/x86_cpu_L3_cache.py

```diff
@@ -0,0 +1,83 @@
+import re
+import logging
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_qemu
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check L3 cache present to guest
+
+    1. boot guest with latest machine_type,
+     checking L3 cache presents inside guest.
+    2. Boot guest with old machine type(rhel7.3.0),
+     L3 cache shouldn't present inside guest.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def boot_and_check_guest(machine_type, check_L3=False):
+        """
+        Boot guest and check L3 cache inside guest
+
+        :param machine_type: Boot guest with which machine type
+        :param check_L3: if L3 cache should exist on guest
+        """
+        params['machine_type'] = machine_type
+        params['start_vm'] = 'yes'
+        vm_name = params['main_vm']
+        L3_existence = 'present' if check_L3 else 'not present'
+        logging.info('Boot guest with machine type %s and expect L3 cache %s'
+                     ' inside guest' % (machine_type, L3_existence))
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        session = vm.wait_for_login()
+        output = session.cmd_output('lscpu')
+        session.close()
+        vm.destroy()
+        L3_present = 'L3 cache' in output
+        if check_L3 ^ L3_present:
+            test.fail('L3 cache should %s inside guest for machine type %s' %
+                      (L3_existence, machine_type))
+
+    def check_version(latest_machine):
+        """
+        Check if the latest supported machine type is newer than the defined
+        old machine type, cancel the test if not.
+
+        :param latest_machine: The latest machine type
+        """
+        latest_ver = re.findall(r'\d+\.\d+', latest_machine)[0].split('.')
+        old_ver = re.findall(r'\d+\.\d+', old_machine)[0].split('.')
```

Comment:
How about that we compare the qemu version string in here?
I mean this :
```
        latest_ver = re.findall(r'\d+\.\d+', latest_machine)[0]
        old_ver = re.findall(r'\d+\.\d+', old_machine)[0]
        if latest_ver <= old_ver:
            test.cancel('The latest supported machine type does not support this test case.')
```
If there is any risk that I haven't considered in this way please let me know, thanks.